File: Functions/get_R_squared.R
GetRSquared <- function(X, W, Y){
  # X: predictor matrix
  # W: regression weights
  # Y: response matrix
  
  Y <- as.matrix(Y)
  Rsq <- sapply(1:ncol(Y), function(i){ 
    var.pred <- sum((as.matrix(X)%*%W[, i] - Y[, i])^2)
    var.y <- sum((Y[,i] - mean(Y[,i]))^2)
    val <- 1 - (var.pred/var.y)
    return(val)
  })
  return(Rsq)
}

File: PyFunctions/get_R_squared.py
import numpy as np

def GetRSquared(X, W, Y):
  '''
  X: predictor matrix
  W: regression weights
  Y: response matrix
  '''
  
  Y = np.asmatrix(Y)
  return 1 - (np.sum((Y - X.dot(W))**2) / np.sum((Y - np.mean(Y))**2))
